kanidmd/src/lib/entry.rs

```diff
@@ -1575,21 +1580,26 @@ impl<VALID, STATE> Entry<VALID, STATE> {
         let v = self
             .attrs
             .entry(AttrString::from(attr))
-            .or_insert_with(Set::new);
+            .or_insert_with(ValueSet::new);
         // Here we need to actually do a check/binary search ...
         v.insert(value);
         // Doesn't matter if it already exists, equality will replace.
     }
 
     /// Overwrite the current set of values for an attribute, with this new set.
-    pub fn set_ava_int(&mut self, attr: &str, values: Set<Value>) {
+    // pub fn set_ava_int(&mut self, attr: &str, values: Set<Value>) {
+    pub fn set_ava_int<T>(&mut self, attr: &str, iter: T)
```

Comment:
Why we pass iter here but not BTreeSet?

Reply:
Because in the future the idea is valueset will change from btreeset internally to a strongly typed interior with different possible representations.